### PR TITLE
Migrate from the silver searcher to ripgrep

### PR DESCRIPTION
**Pull request by stevenbarragan**
ripgrep is faster than {grep, ag, git grep, ucg, pt, sift}.

check this [blog post](http://blog.burntsushi.net/ripgrep/).

### CODE REVIEW

**Comment by lfilho** [comment]
@skwp this seems super nice. Any thoughts?

@stevenbarragan the branch has conflicts now :/

**Comment by stevenbarragan** [comment]
@lfilho merge conflicts fixed.

**Comment by stevenbarragan** [comment]
@skwp any thoughts about this PR? or maybe you need some help maintaining the repo? I've seen some PR's stay open for ever

**Comment by skwp** [comment]
Sorry guys, I've really been unable to do a good job maintaining. I'm revisiting the PRs

**Comment by skwp** [comment]
My main issue with this it will break a lot of people who don't have it brew installed and just pull on the repo. Why not provide this as a new mapping `,rg`?

File: vim/settings/ctrlp.vim [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    \ 'ag %s --files-with-matches -g "" --ignore "\.git$\|\.hg$\|\.svn$"'
 
-  " ag is fast enough that CtrlP doesn't need to cache
+if executable('rg')
+  " Use rg in CtrlP for listing files. Lightning fast and respects .gitignore
+  let g:ctrlp_user_command = 'rg --files --smart-case %s'
+
+  " rg is fast enough that CtrlP doesn't need to cache
   let g:ctrlp_use_caching = 0
 else
```
Comment (skwp) [inline vim/settings/ctrlp.vim]: i would do a fallback to ag here

**Comment by stevenbarragan** [comment]
@skwp I added the fall back to `ag`, and I added the mapping `,rg` in [here](https://github.com/skwp/dotfiles/pull/738/files#diff-36a0812e9ffccd78c05cc904bd17f9b6).

**Comment by templeman** [comment]
Also worth noting that ag.vim has been deprecated: https://github.com/rking/ag.vim/issues/124#issuecomment-227038003.

**Comment by lfilho** [comment]
Speaking of backwards compatibilities, should we try to avoid key mappings too specific for plugins? For example, instead of `,ag` and `,rg`, why not have only something like `,fa` (*f*ind in *a*ll) or `,fe` (*f*ind *e*verywhere) or whatever?

This way we would be less dependent and coupled with which plugins provide certain features...

**Comment by skwp** [comment]
We do have a generic mapping of `,gg` also (go grep, git grep, or whatever mnemonic you like there)

**Comment by skwp** [comment]
Sorry guys I had to revert, it turns out ripgrep doesn't install cleanly on my machine. It tried to compile but failed. Until I understand that I dont' want to break other people

**Comment by skwp** [comment]
```
configure: error: in `/private/tmp/rust-20170530-4679-511zir/rustc-1.17.0-src/build/x86_64-apple-darwin/native/jemalloc':
configure: error: C compiler cannot create executables
See `config.log' for more details

Build failed, waiting for other jobs to finish...
error: build failed


command did not execute successfully: "/private/tmp/rust-20170530-4679-511zir/rustc-1.17.0-src/build/x86_64-apple-darwin/stage0/bin/cargo" "build" "-j" "8" "--target" "x86_64-apple-darwin" "--release" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/private/tmp/rust-20170530-4679-511zir/rustc-1.17.0-src/src/libstd/Cargo.toml"
expected success, got: exit code: 101
```

**Comment by lfilho** [comment]
That's weird. I installed via brew and it works fine.

**Comment by stevenbarragan** [comment]
I'm sorry about that, I didn't have any issue either, I'll take a look.

**Comment by lfilho** [comment]
@skwp sanity check: have you update your brew recently?

I'm running brew 1.2.1, OSX El Capitan and zsh 5.3.1

**Comment by lfilho** [comment]
@stevenbarragan and @skwp did you have a chance to have a look? I think this PR will greatly speed things for everyone in #661

**Pull request merged**